ios/FluentUI/Vnext/Drawer/SlideOverPanel.swift

```diff
@@ -12,84 +12,137 @@ import SwiftUI
                 MSFInteractiveSpacer(defaultBackgroundColor: $tokens.backgroundClearColor)
                     .onTapGesture (perform: actionOnBackgroundTap ?? {})
             }
         }
-        .background(isOpen ? tokens.backgroundDimmedColor.opacity(backgroundLayerOpacity) : tokens.backgroundClearColor)
+        .background(resolvedBackgroundColor)
     }
 
-    private func resolvedContentOffset() -> CGFloat {
-        // override offset if required
-        if let preferredContentWidth = preferredContentOffset {
-            // parent view wants to take over primarily to conform user drag gesture
-            return preferredContentWidth
+    private let contentWidthSizeRatio: CGFloat = 0.9
+
+    /// Read-only property for contentWidth to resize content size use `contentWidthSizeRatio` instead
+    private var contentWidth: CGFloat {
+        return slideOutPanelWidth * contentWidthSizeRatio
+    }
+
+    private var resolvedContentOffset: CGFloat {
+
+        var offset: CGFloat
+        switch transitionState {
+        case .collapsed:
+            offset = collapsedContentOffset
+        case .expanded:
+            offset = expandedContentOffset
+        case .inTransisiton:
+            offset = percentTransistionOffset
         }
 
-        let offset = isOpen ? expandedConentOffset() : collapsedContentOffset()
         if direction == .left {
             return -offset
         } else {
             return offset
         }
     }
 
-    private func expandedConentOffset() -> CGFloat {
+    private var resolvedBackgroundColor: Color {
+        guard backgroundDimmed else {
+            return tokens.backgroundClearColor
+        }
+        let opacity = resolvedPanelStateValue(tokens.backgroundClearOpacity, tokens.backgroundDimmedOpacity)
+        return tokens.backgroundDimmedColor.opacity(opacity)
+    }
+
+    private var resolvedShadowColor: Color {
+        let opacity = resolvedPanelStateValue(0.0, tokens.shadowOpacity)
+        return tokens.shadowColor.opacity(opacity)
+    }
+
+    private var percentTransistionOffset: CGFloat {
+        // Override offset if required
+        if let percentDriveTransition = percentTransition {
+            // parent view wants to take over primarily to conform user drag gesture
+            if percentDriveTransition >= 0 && percentDriveTransition <= 1 {
+                return expandedContentOffset + collapsedContentOffset * CGFloat(1 - percentDriveTransition)
+            }
+        }
+        return CGFloat.zero
+    }
+
+    private var expandedContentOffset: CGFloat {
         return CGSize.zero.width
     }
 
-    private func collapsedContentOffset() -> CGFloat {
+    private var collapsedContentOffset: CGFloat {
         return slideOutPanelWidth * contentWidthSizeRatio
     }
 
-    private func contentWidth() -> CGFloat {
-        return slideOutPanelWidth * contentWidthSizeRatio
+    private func resolvedPanelStateValue(_ clear: Double, _ opaque: Double) -> Double {
```

Comment:
I like the idea of having a generic function to calculate values based on the percentage, but the parameters should be consistent on the generic name. So, can you rename the parameters from clear to **minValue** and opaque to **maxValue**?